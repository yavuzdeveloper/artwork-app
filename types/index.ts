export type Artwork = {
  id: string;
  title: string;
  image_id: string;
  iiif_url: string;
  description?: string;
};

export interface Pagination {
  total: number;
  total_pages: number;
  current_page: number;
}

export interface ArtAPIResponse {
  data: Artwork[];
  pagination: Pagination;
  config: {
    iiif_url: string;
  };
}
