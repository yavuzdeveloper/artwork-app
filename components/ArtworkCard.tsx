import Link from "next/link";
import { Artwork } from "@/types";

export default function ArtworkCard({ artwork }: { artwork: Artwork }) {
  return (
    <Link
      href={`/artwork/${artwork.id}`}
      className="group relative block overflow-hidden rounded-xl bg-white shadow-md hover:shadow-xl transition-all duration-300 ease-in-out transform hover:-translate-y-1"
    >
      <div className="relative aspect-square overflow-hidden">
        <img
          src={`${artwork.iiif_url}/${artwork.image_id}/full/843,/0/default.jpg`}
          alt={artwork.title}
          className="w-full h-full object-cover transition-transform duration-500 group-hover:scale-105"
          loading="lazy"
        />
        <div className="absolute inset-0 bg-gradient-to-t from-black/30 to-transparent opacity-0 group-hover:opacity-100 transition-opacity duration-300" />
      </div>
      <div className="p-5">
        <h2 className="text-xl font-semibold text-gray-800 line-clamp-2">
          {artwork.title}
        </h2>
      </div>
    </Link>
  );
}
