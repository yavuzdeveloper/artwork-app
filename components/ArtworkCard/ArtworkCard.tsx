import Link from "next/link";

import { Artwork } from "@/types";
import GridImage from "../GridImage/GridImage";

export default function ArtworkCard({ artwork }: { artwork: Artwork }) {
  return (
    <Link
      href={`/artwork/${artwork.id}`}
      className="group relative block overflow-hidden rounded-xl bg-white shadow-md hover:shadow-xl transition-all duration-300 ease-in-out transform hover:-translate-y-1"
    >
      <div className="relative aspect-square overflow-hidden bg-gray-100">
        <GridImage
          artwork={artwork}
          className="w-full h-full object-cover transition-transform duration-500 group-hover:scale-105"
        />
        <div className="absolute inset-0 bg-gradient-to-t from-black/30 to-transparent opacity-0 group-hover:opacity-100 transition-opacity duration-300" />
      </div>
      <div className="p-5">
        <h3 className="text-lg font-semibold text-gray-800 line-clamp-2">
          {artwork.title}
        </h3>
      </div>
    </Link>
  );
}
