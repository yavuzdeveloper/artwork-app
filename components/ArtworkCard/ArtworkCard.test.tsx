import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import ArtworkCard from "./ArtworkCard";
import { Artwork } from "@/types";

// Mock next/link for Next.js 13+
jest.mock("next/link", () => ({
  __esModule: true,
  default: ({
    children,
    href,
  }: {
    children: React.ReactNode;
    href: string;
  }) => <a href={href}>{children}</a>,
}));

// Mock GridImage component
jest.mock("../GridImage/GridImage", () => ({
  __esModule: true,
  default: ({
    artwork,
    className,
  }: {
    artwork: Artwork;
    className: string;
  }) => (
    <img
      src={`${artwork.iiif_url}/${artwork.image_id}/full/843,/0/default.jpg`}
      alt={artwork.title}
      className={className}
      data-testid="grid-image"
    />
  ),
}));

const mockArtwork: Artwork = {
  id: "1",
  title: "Mona Lisa",
  image_id: "1",
  iiif_url: "https://iiif.artic.edu/iiif/2",
  description: "",
};

describe("ArtworkCard Component", () => {
  it("should render correctly with artwork data", () => {
    render(<ArtworkCard artwork={mockArtwork} />);

    // Verify title is displayed
    expect(screen.getByText("Mona Lisa")).toBeInTheDocument();

    // Verify link has correct href
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", "/artwork/1");

    // Verify image is rendered with correct props
    const image = screen.getByTestId("grid-image");
    expect(image).toHaveAttribute(
      "src",
      "https://iiif.artic.edu/iiif/2/1/full/843,/0/default.jpg"
    );
    expect(image).toHaveClass("w-full h-full object-cover");
  });

  it("should have correct hover effect classes", () => {
    render(<ArtworkCard artwork={mockArtwork} />);

    // Verify image hover effect
    const image = screen.getByTestId("grid-image");
    expect(image).toHaveClass("group-hover:scale-105");

    // Verify overlay hover effect
    const overlay = document.querySelector(".bg-gradient-to-t");
    expect(overlay).toHaveClass("opacity-0");
    expect(overlay).toHaveClass("group-hover:opacity-100");
  });

  it("should maintain proper aspect ratio", () => {
    render(<ArtworkCard artwork={mockArtwork} />);
    const imageContainer = document.querySelector(".aspect-square");
    expect(imageContainer).toBeInTheDocument();
  });
});
