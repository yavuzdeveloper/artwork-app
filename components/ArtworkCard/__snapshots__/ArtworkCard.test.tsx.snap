// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkCard Component matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/artwork/1"
  >
    <div
      class="relative aspect-square overflow-hidden bg-gray-100"
    >
      <img
        alt="Mona Lisa"
        class="w-full h-full object-cover transition-transform duration-500 group-hover:scale-105"
        data-testid="grid-image"
        src="https://iiif.artic.edu/iiif/2/1/full/843,/0/default.jpg"
      />
      <div
        class="absolute inset-0 bg-gradient-to-t from-black/30 to-transparent opacity-0 group-hover:opacity-100 transition-opacity duration-300"
      />
    </div>
    <div
      class="p-5"
    >
      <h3
        class="text-lg font-semibold text-gray-800 line-clamp-2"
      >
        Mona Lisa
      </h3>
    </div>
  </a>
</DocumentFragment>
`;
